SearchEngine/lambda_SearchEngine/include/log4pp.h

```diff
@@ -0,0 +1,121 @@
+#pragma once
+#include <iostream>
+#include <string>
+#include <log4cpp/SimpleLayout.hh>
+#include <log4cpp/BasicLayout.hh>
+#include <log4cpp/PatternLayout.hh>
+#include <log4cpp/OstreamAppender.hh>
+#include <log4cpp/FileAppender.hh>
+#include <log4cpp/RollingFileAppender.hh>
+#include <log4cpp/Category.hh>
+#include <log4cpp/Priority.hh>
+
+using namespace std;
+#define suffix(msg) std::string(msg).append(" ##")\
+        .append(__FILE__).append(":").append(__func__)\
+        .append(":").append(int2string(__LINE__))\
+        .append("##").c_str()
+
+#define logError(msg, ...) Mylogger::getInstance()->error(suffix(msg),##__VA_ARGS__)
+#define logWarn(msg, ...) Mylogger::getInstance()->warn(suffix(msg),##__VA_ARGS__)
+#define logInfo(msg, ...) Mylogger::getInstance()->info(suffix(msg),##__VA_ARGS__)
+#define logDebug(msg, ...) Mylogger::getInstance()->debug(suffix(msg),##__VA_ARGS__)
+
+inline std::string int2string(int line) {
+    std::ostringstream oss;
+    oss << line;
+    return oss.str();
+}
+
+using std::cout;
+using std::endl;
+using namespace log4cpp;
+
+
+class Mylogger
+{
+public:
+
+    template<class...Args>
+	void warn(const char * msg,Args...args);
+
+    template<class...Args>
+	void error(const char * msg,Args...args);
+
+    template<class...Args>
```

Comment:
use typename insteal of class to define a template type.
class is a bit confusiong.